python/dgl/nn/pytorch/explain/subgraphx.py

```diff
@@ -364,3 +365,453 @@ def explain_graph(self, graph, feat, target_class, **kwargs):
+    Explorations <https://arxiv.org/abs/2102.05152>`__, adapted for heterogeneous graphs
+
+    It identifies the most important subgraph from the original graph that
+    plays a critical role in GNN-based graph classification.
+
+    It employs Monte Carlo tree search (MCTS) in efficiently exploring
+    different subgraphs for explanation and uses Shapley values as the measure
+    of subgraph importance.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat)`.
+        * The output of its forward function is the logits.
+    num_hops : int
+        Number of message passing layers in the model
+    coef : float, optional
+        This hyperparameter controls the trade-off between exploration and
+        exploitation. A higher value encourages the algorithm to explore
+        relatively unvisited nodes. Default: 10.0
+    high2low : bool, optional
+        If True, it will use the "High2low" strategy for pruning actions,
+        expanding children nodes from high degree to low degree when extending
+        the children nodes in the search tree. Otherwise, it will use the
+        "Low2high" strategy. Default: True
+    num_child : int, optional
+        This is the number of children nodes to expand when extending the
+        children nodes in the search tree. Default: 12
+    num_rollouts : int, optional
+        This is the number of rollouts for MCTS. Default: 20
+    node_min : int, optional
+        This is the threshold to define a leaf node based on the number of
+        nodes in a subgraph. Default: 3
+    shapley_steps : int, optional
+        This is the number of steps for Monte Carlo sampling in estimating
+        Shapley values. Default: 100
+    log : bool, optional
+        If True, it will log the progress. Default: False
+    """
+
+    def __init__(
+        self,
+        model,
+        num_hops,
+        coef=10.0,
+        high2low=True,
+        num_child=12,
+        num_rollouts=20,
+        node_min=3,
+        shapley_steps=100,
+        log=False,
+    ):
+        super().__init__()
+        self.num_hops = num_hops
+        self.coef = coef
+        self.high2low = high2low
+        self.num_child = num_child
+        self.num_rollouts = num_rollouts
+        self.node_min = node_min
+        self.shapley_steps = shapley_steps
+        self.log = log
+
+        self.model = model
+
+    def shapley(self, subgraph_nodes):
+        r"""Compute Shapley value with Monte Carlo approximation.
+
+        Parameters
+        ----------
+        subgraph_nodes : dict[str, Tensor]
+            The dictionary that associates tensor node ids (values) with
+            respective node types (keys) present in graph which are associated
+            with this tree node.
```

Comment:
subgraph_nodes[nty] gives the tensor node IDs of node type nty in the subgraph, which are associated with this tree node